server.go

```diff
@@ -116,8 +129,52 @@ func (s *server) Start() {
 	s.fundingMgr.Start()
 	s.htlcSwitch.Start()
 
+	// ROUTING ADDED
+	s.routingMgr.Start()
+
 	s.wg.Add(1)
 	go s.queryHandler()
+
+	// ROUTING ADDED
+	s.wg.Add(1)
+	go s.routingOutHandler()
+}
+
+// ROUTING ADDED
+// Handles outgoing routing messages
+// Determines correct peer and forward message to it
+func (s *server) routingOutHandler(){
+	// TODO (mkl) Add copying
+	out:
+	for {
+		select {
+		// Find required peer
+		case msg :=<- s.routingMgr.ChOut:
+			msg1 := msg.(routingmessages.Message)
+			receiverID := msg1.GetReceiverID().ToByte32()
+			var targetPeer *peer
+			for _, peer := range s.peers { // TODO: threadsafe api
+				// We found the the target
+				if peer.lightningID == receiverID {
+					targetPeer = peer
+					break
+				}
+			}
+			if targetPeer != nil{
+				fmt.Println("Peer found. Sending message")
+				msg2 := outgoinMsg{
+					msg: msg.(lnwire.Message),
+					sentChan: make(chan struct{}, 1),
+				}
+				targetPeer.outgoingQueue <- msg2
```

Comment:
Sure, the peer struct has a [`queueMsg`](https://github.com/lightningnetwork/lnd/blob/master/peer.go#L534) method creates and sends the `outgoingMsg` directly to the peer.